### PR TITLE
Added ban reason and scope for banned competitors

### PR DESCRIPTION
**Pull request by danieljames-dj**
This was a request from WDC, hence implementing this

### CODE REVIEW

File: app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitors.jsx [hunk 1]
```diff
@@ -20,6 +20,8 @@ export default function BannedCompetitors({
             <Table.HeaderCell width={5}>User</Table.HeaderCell>
             <Table.HeaderCell width={2}>Start date</Table.HeaderCell>
             <Table.HeaderCell width={2}>End date</Table.HeaderCell>
+            <Table.HeaderCell width={2}>Ban reason</Table.HeaderCell>
+            <Table.HeaderCell width={2}>Ban scope</Table.HeaderCell>
```
Comment (kr-matthews) [inline app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitors.jsx:24]: For consistency with the other header cells, eg "Start date" means "Ban start date" already:
```suggestion
            <Table.HeaderCell width={2}>Start date</Table.HeaderCell>
            <Table.HeaderCell width={2}>End date</Table.HeaderCell>
            <Table.HeaderCell width={2}>Reason</Table.HeaderCell>
            <Table.HeaderCell width={2}>Scope</Table.HeaderCell>
```

File: app/controllers/api/v0/user_roles_controller.rb [hunk 2]
```diff
@@ -166,6 +170,19 @@ def create
     changed_value.nil? ? 'None' : changed_value
   end
 
+  private def push_previous_changes_to_changes(previous_changes, changes)
```
Comment (gregorbg) [inline app/controllers/api/v0/user_roles_controller.rb:173]: I am not a big fan of manipulating (writing to) values that are passed as parameter.
This method should just return a "friendly" version of `previous_changes`, i.e. map it to `UserRoleChange` values and then whoever is using the return value from this method can append it to `changes`.

File: app/models/roles_metadata_banned_competitors.rb [hunk 3]
```diff
@@ -1,4 +1,8 @@
 # frozen_string_literal: true
 
 class RolesMetadataBannedCompetitors < ApplicationRecord
+  enum :scope, {
+    competing_only: "competing_only",
+    competing_and_attending: "competing_and_attending",
```
Comment (gregorbg) [inline app/models/roles_metadata_banned_competitors.rb:6]: I'm assuming you discussed this with WDC, but what about other scopes like `organizing` or `delegating`?

File: app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitors.jsx [hunk 4]
```diff
@@ -20,6 +20,8 @@ export default function BannedCompetitors({
             <Table.HeaderCell width={5}>User</Table.HeaderCell>
             <Table.HeaderCell width={2}>Start date</Table.HeaderCell>
             <Table.HeaderCell width={2}>End date</Table.HeaderCell>
+            <Table.HeaderCell width={2}>Reason</Table.HeaderCell>
+            <Table.HeaderCell width={2}>Scope</Table.HeaderCell>
```
Comment (gregorbg) [inline app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitors.jsx:24]: SemUI widths should ideally add up to 16

**Review by gregorbg — CHANGES_REQUESTED** [review]

File: app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitors.jsx (hunk 4 above)
Comment (danieljames-dj) [reply app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitors.jsx]: Oh I didn't knew it, done.

File: app/models/roles_metadata_banned_competitors.rb (hunk 3 above)
Comment (danieljames-dj) [reply app/models/roles_metadata_banned_competitors.rb:6]: As of now WDC asked for only these two scopes and they might increase them in future.

File: app/controllers/api/v0/user_roles_controller.rb (hunk 2 above)
Comment (danieljames-dj) [reply app/controllers/api/v0/user_roles_controller.rb]: Okay, I've made the method return the changes list and appended it from where it's called.

File: app/controllers/api/v0/user_roles_controller.rb [hunk 5]
```diff
@@ -166,6 +170,21 @@ def create
     changed_value.nil? ? 'None' : changed_value
   end
 
+  private def push_previous_changes_to_changes(previous_changes)
```
Comment (gregorbg) [inline app/controllers/api/v0/user_roles_controller.rb:173]: The name of this method is no longer accurate after your recent changes.

File: app/controllers/api/v0/user_roles_controller.rb [hunk 6]
```diff
@@ -166,6 +170,21 @@ def create
...
+    changes = []
```
Comment (gregorbg) [inline app/controllers/api/v0/user_roles_controller.rb:174]: Just return `previous_changes&.map do ... end` directly

**Review by gregorbg — CHANGES_REQUESTED** [review]

File: app/controllers/api/v0/user_roles_controller.rb (hunk 5 above)
Comment (danieljames-dj) [reply app/controllers/api/v0/user_roles_controller.rb]: Oh ya, changed

File: app/controllers/api/v0/user_roles_controller.rb (hunk 6 above)
Comment (danieljames-dj) [reply app/controllers/api/v0/user_roles_controller.rb]: Done

**Review by gregorbg — APPROVED** [review]

**Pull request merged**